components/variations/BUILD.gn

```diff
@@ -1,12 +1,22 @@
+# Copyright (c) 2021 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
 declare_args() {
   brave_variations_server_url = ""
+  brave_variations_insecure_server_url = ""
 }
 
-if (is_official_build) { assert(brave_variations_server_url != "") }
+if (is_official_build) {
+  assert(brave_variations_server_url != "" &&
+         brave_variations_insecure_server_url != "")
+}
 
 config("brave_variations_config") {
   defines = [
-    "BRAVE_VARIATIONS_SERVER_URL=\"$brave_variations_server_url\""
+    "BRAVE_VARIATIONS_SERVER_URL=\"$brave_variations_server_url\"",
+    "BRAVE_VARIATIONS_INSECURE_SERVER_URL=\"$brave_variations_insecure_server_url\"",
```

Comment:
do we really need that? I think the plan was to do nothing if the normal (aka secure) URL is not accessible. Probably it broke at some point (cc @moritzhaller @diracdeltas )

Reply:
@iefremov How about removing the definitions from GN files, and then setting `kVariationsInsecureServerURL` switch in `brave_main_delegate.cc` to `BRAVE_VARIATIONS_SERVER_URL` (secure URL). This way when it gets to `VariationsService::MaybeRetryOverHTTP` and checks for `kHttpScheme` it'll return false and won't try.

Reply:
sounds good

Reply:
Updated in 686bd52